### PR TITLE
Provide DDoc comments for globals.d

### PR DESCRIPTION
**Pull request by dkorpel**
I'm trying out adding DDoc comments to public symbols in DMD after this thread on the forum: [Where are the the comments in dmd?](https://forum.dlang.org/post/ujfkfzkizgdhexigogmw@forum.dlang.org)

Initially I made the fields in the `Params` struct DDoc comments as well, but that gave a large diff with huge addition/deletion blocks so I'm leaving those alone for now.

I did not edit `globals.h`, if I recall correctly the file will eventually get superseded by the automatically generated `frontend.h`, so I'm not sure if it's useful to keep the comments in there in sync with the .d file.

### CODE REVIEW

**Review by UplinkCoder — APPROVED** [review]
nice

**Comment by thewilsonator** [comment]
```
[1msrc/dmd/globals.d(537): [1;33mWarning: [mDdoc: Stray ')'. This may cause incorrect Ddoc output. Use $(RPAREN) instead for unpaired right parentheses.
[1msrc/dmd/globals.d(537): [1;33mWarning: [mDdoc: Stray ')'. This may cause incorrect Ddoc output. Use $(RPAREN) instead for unpaired right parentheses.
```

File: src/dmd/globals.d [hunk 1]
```diff
@@ -269,38 +275,50 @@ enum hdr_ext  = "di";       // for D 'header' import files
...
+    Array!Identifier* debugids;   /// command line debug versions and predefined versions
 
-    enum recursionLimit = 500; // number of recursive template expansions before abort
+    enum recursionLimit = 500; /// number of recursive template expansions before abort
 
   nothrow:
 
-    /* Start gagging. Return the current number of gagged errors
+    /**
+     * Start ignoring compile errors instead of reporting them
```
Comment (MoonlightSentinel) [inline src/dmd/globals.d:313]:
```suggestion
     * Start ignoring compile errors instead of reporting them.
```

File: src/dmd/globals.d [hunk 2]
```diff
@@ -309,8 +327,11 @@ extern (C++) struct Global
         return gaggedErrors;
     }
 
-    /* End gagging, restoring the old gagged state.
-     * Return true if errors occurred while gagged.
+    /**
+     * End gagging, restoring the old gagged state.
```
Comment (MoonlightSentinel) [inline src/dmd/globals.d:331]:
```suggestion
     * Stop gagging, restoring the old gagged state before the most recent call to `startGagging`.

```

File: src/dmd/globals.d [hunk 3]
```diff
@@ -323,9 +344,10 @@ extern (C++) struct Global
         return anyErrs;
     }
 
-    /*  Increment the error count to record that an error
-     *  has occurred in the current context. An error message
-     *  may or may not have been printed.
+    /**
+     * Increment the error count to record that an error has occurred in the current context.
```
Comment (MoonlightSentinel) [inline src/dmd/globals.d:348]: Which context?

File: src/dmd/globals.d [hunk 4]
```diff
@@ -509,10 +537,9 @@ nothrow:
         return buf.extractChars();
     }
 
-    /* Checks for equivalence,
-     * a) comparing the filename contents (not the pointer), case-
-     *    insensitively on Windows, and
-     * b) ignoring charnum if `global.params.showColumns` is false.
+    /**
+     * Checks for equivalence, comparing the filename contents (not the pointer) case-
+     * insensitively on Windows, and ignoring `charnum` if `global.params.showColumns` is false.
```
Comment (MoonlightSentinel) [inline src/dmd/globals.d:542]: The current phrasing is rather convoluted, maybe try something like this:

```suggestion
     * Checks for equivalence by comparing the filename contents (not the pointer) and character location.
     *
     * Note:
     *  - Uses case-insensitive comparison on Windows
     *  - Ignores `charnum` if `global.params.showColumns` is false.
```

File: src/dmd/globals.d [hunk 5]
```diff
@@ -569,28 +601,34 @@ enum LINK : ubyte
     system,
 }
 
+/// Whether to mangle an external aggregate as a struct or class, as set by `extern(C++, struct)`
 enum CPPMANGLE : ubyte
 {
-    def,
+    def, /// default
     asStruct,
     asClass,
```
Comment (MoonlightSentinel) [inline src/dmd/globals.d:609]:
```suggestion
    def,         /// default
    asStruct, /// `extern(C++, struct)`
    asClass,  /// `extern(C++, class)`
```

**Review by MoonlightSentinel — APPROVED** [review]
LGTM except some minor nitpicks.

File: src/dmd/globals.d (hunk 3 above)
Comment (dkorpel) [reply src/dmd/globals.d:348]: I'm not sure actually, I didn't write the original comment. It's only used once for expression semantic of a `StructLiteralExp` when filling the remaining fields fails.

**Pull request merged**